### PR TITLE
Index

### PR DESCRIPTION
**Pull request by reginafcompton**
@hancush - this PR seems bigger than it is. Nonetheless, I should have broken it into smaller PRs – the JS refactor encompassed more than I had planned. In any case, here's what the PR entails:
* undertake a JS refactor (i.e., consolidating three JS files, which did essentially the same thing, into one)
* isolate unneeded templates into `reserved-templates` (these can be deleted, eventually)
* move third-party JS libraries into a `lib` folder 
* make data about sold lots for use in Carto

I would like you to focus on the contents of this file, which explains the steps I took to make the Carto data: https://github.com/datamade/large-lots/pull/284/files#diff-c40ef20bda0d52ead22830db63bc3b43 I'd like to wrap this inside a Make file, but I am thinking that should occur in a different PR, following your input on this.

### CODE REVIEW

File: data/pilots_all/DATA_PREP.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+UPDATE lots_sold_all_pilots SET status='sold';
+```
+
+### Create a dataset for Carto
+
+Query your database to create a file with lot PINs, addresses, communities, wards, and geoms.
+
+```
+COPY (
+  SELECT pin14 as pin_nbr, pins.pin, ST_AsGeoJSON(ST_transform(pins.geom, 4326)) as the_geom, low_address, high_address, street_direction, street_name, street_type, community, ward, status
```
Comment (hancush) [inline data/pilots_all/DATA_PREP.md]: Can you add some line breaks here to make this query easier to parse? i.e.

```sql
COPY(
  SELECT
    pin14 as pin_nbr,
    pins.pin,
    ...
  FROM (
    select
      pin14,
      ...
  ) AS pins
  INNER JOIN ...
) TO 'foo' CSV HEADER

File: data/pilots_all/DATA_PREP.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    on trim(replace("PIN number", '-', ''))=parcel.pin10
+    ) as pins
+  INNER JOIN community_areas
+  ON ST_Intersects(pins.geom, community_areas.geom)
+  INNER JOIN wards 
+  ON ST_Intersects(pins.geom, wards.geom)
+  LEFT JOIN raw_address
+  ON pin14=replace(raw_address.pin, '-', '')
+) TO '<path to repo>/large-lots/data/pilots_all/all_sold_lots.csv' 
+DELIMITER ',' CSV HEADER;
```
Comment (hancush) [inline data/pilots_all/DATA_PREP.md]: DELIMITER is `,` by default in CSV mode so you don't need to specify here. (I learned this last week!)

File: templates/reserved-templates/status_pilot_1.html [hunk 3]
```diff
@@ -2,8 +2,8 @@
 {% load staticfiles %}
```
[the remark this answers is not in the document]
Comment (reginafcompton) [reply templates/reserved-templates/status_pilot_1.html]: They. Are. Gone.

File: data/pilots_all/DATA_PREP.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# Preparing the Data
+
+The index page displays all the properties sold and currently under review. Getting that data requires a little effort: the following explains what you need to do.
+
+### Process data sources 
+
+First, you need a Postgress database with all city parcels, communities, and wards. Please see `DATA_PREP.md` at the root of the large-lots repo for information on creating a database with the necessary tables. Alternatively, you may set-up an `lis` database, given the instructions provided in [this repo](https://github.com/datamade/lis).
+
+Then, request a list of all properties sold in previous pilots from the Department of Planning and Development (contact: Jeanne Chandler). Add that file to `pilots_all/raw/`, and transform it into a Postgres table using csvsql.
```
[the remark this answers is not in the document]
Comment (reginafcompton) [reply data/pilots_all/DATA_PREP.md]: I'd like to put it under version control, given that acquiring this file necessitates emailing and waiting on Jeanne. (It's not just a link we can wget, for instance.) If someone needs the raw data, it's there.

File: templates/reserved-templates/status_pilot_1.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (hancush) [reply templates/reserved-templates/status_pilot_1.html]: rip in peace 

skeleton rejoices

File: data/pilots_all/Makefile [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# Variables
+PG_DB=lis
+SOLD_LOTS=raw/Large\ Lot\ Sold\ GHN\ EGP\ and\ Austin\ Roseland\ and\ Auburn\ Gresham.csv
+
+# Non-file targets
+.PHONY: clean all
+
+clean:
+		rm -Rf processed/*
```
Comment (hancush) [inline data/pilots_all/Makefile]: I'd just do `rm processed/*` to delete the files in the directory without deleting the directory.

File: data/pilots_all/Makefile [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+# Create table with necessary columns in database
+lots_sold_all_pilots.table: $(SOLD_LOTS)
+		csvsql --db postgresql:///$(PG_DB) --insert $(SOLD_LOTS) --table lots_sold_all_pilots
+		psql -d $(PG_DB) -c "ALTER TABLE lots_sold_all_pilots ADD COLUMN status varchar(30)"
+		psql -d $(PG_DB) -c "UPDATE lots_sold_all_pilots SET status='sold'"
+		touch processed/$@
+
+# Query database and make a csv
+all_sold_lots.csv:
```
Comment (hancush) [inline data/pilots_all/Makefile]: `lots_sold_all_pilots.table` is a dependency of this target, so you'll want to add it here.

File: data/pilots_all/Makefile [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+          ON trim(replace(\"PIN number\", '-', ''))=parcel.pin10 \
+      ) as pins \
+      INNER JOIN community_areas \
+      ON ST_Intersects(pins.geom, community_areas.geom) \
+      INNER JOIN wards \
+      ON ST_Intersects(pins.geom, wards.geom) \
+      LEFT JOIN raw_address \
+      ON pin14=replace(raw_address.pin, '-', '') \
+    ) TO STDOUT WITH CSV HEADER" > processed/$@
+		touch processed/$@
```
Comment (hancush) [inline data/pilots_all/Makefile]: You don't need to `touch` the file here. It's used in the recipe for the database table as a placeholder; Make doesn't know your database table has been created, but you can `touch blah.table` at the end of your recipe for the table to signal to recipes that may depend on it that it exists. By contrast, this recipe creates an actual CSV file, so no placeholder is necessary. Does that make sense?

Comment (reginafcompton) [reply data/pilots_all/Makefile]: Oh, yes. The `touch` at the end here is not very useful, is it.

File: data/pilots_all/Makefile (hunk 6 above)
Comment (hancush) [reply data/pilots_all/Makefile]: I know I said `lots_sold_all_pilots.table`, but we `touch processes/lots_sold_all_pilots.table` in the recipe, so we need to include the dir in the dependency. Sorry for leading you astray! (ha)

Comment (reginafcompton) [reply data/pilots_all/Makefile]: No worries! That, of course, makes more sense.

**Pull request merged**